### PR TITLE
VIS: Add xlabel/ylabel to `plot.hist()`

### LINKED ISSUE #49793: BUG: Can't set xlabel/ylabel on histogram
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
import numpy as np
import pandas as pd

np.random.seed(1)

ax = pd.Series(np.random.randn(1000) * 10 + 60).plot.hist(
    xlabel='grade', ylabel='students', ec='black'
)
```

### Issue Description

`xlabel` and `ylabel` are accepted, but ignored.

### Expected Behavior

As with other plotting methods, I would expect this to label the x and y axes as requested.

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : e660f2cf774c78d0d16b4af153b502a8abe49992
python           : 3.10.6.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 21.6.0
Version          : Darwin Kernel Version 21.6.0: Mon Aug 22 20:19:52 PDT 2022; root:xnu-8020.140.49~2/RELEASE_ARM64_T6000
machine          : arm64
processor        : arm
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.4.0.dev0+3654.ge660f2cf77.dirty
numpy            : 1.23.4
pytz             : 2022.1
dateutil         : 2.8.2
setuptools       : 65.5.0
pip              : 22.2.2
Cython           : 0.29.32
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.9.1
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.1.2
IPython          : 8.6.0
pandas_datareader: None
bs4              : 4.11.1
bottleneck       : None
brotli           : 
fastparquet      : None
fsspec           : None
gcsfs            : None
matplotlib       : 3.6.1
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : None
snappy           : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
zstandard        : None
tzdata           : None
qtpy             : None
pyqt5            : None

</details>

### PR DESCRIPTION
**Pull request by stefmolin**
- [X] closes #49793 
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [N/A] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [X] Added an entry in the latest `doc/source/whatsnew/v2.0.0.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/plotting/_core.py [hunk 1]
```diff
@@ -720,16 +720,24 @@ class PlotAccessor(PandasObject):
...
+            Now applicable to histograms.
+
     ylabel : label, optional
         Name to use for the ylabel on y-axis. Default will show no ylabel, or the
         y-column name for planar plots.
 
         .. versionadded:: 1.1.0
 
-        .. versionchanged:: 1.2.0
+        .. versionchanged:: 2.0.0
```
Comment (mroeschke) [inline pandas/plotting/_core.py:733]: I think this should remain 1.2 since it applies to the scatter/hexbin note below

Comment (stefmolin) [reply pandas/plotting/_core.py]: Good catch &ndash; not sure why this got changed.

File: pandas/tests/plotting/test_series.py [hunk 2]
```diff
@@ -818,6 +818,9 @@ def test_xlabel_ylabel_series(self, kind, index_name, old_label, new_label):
         if kind == "barh":
             assert ax.get_xlabel() == ""
             assert ax.get_ylabel() == old_label
+        elif kind == "hist":
```
Comment (mroeschke) [inline pandas/tests/plotting/test_series.py:821]: I think we need a separate test to check if a user passes a custom xlabel and ylabel

Comment (stefmolin) [reply pandas/tests/plotting/test_series.py:821]: I believe this is covered with lines 829-831, which are not showing up in the diff without expanding it.
Screen Shot 2022-12-03 at 2 27 10 PM

Comment (mroeschke) [reply pandas/tests/plotting/test_series.py:821]: Ah gotcha, thanks!

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @stefmolin

**Pull request merged**